ql/src/java/org/apache/hadoop/hive/ql/hooks/QueryLifeTimeHookContextImpl.java

```diff
@@ -78,11 +88,12 @@ public Builder withHookContext(HookContext hc) {
       return this;
     }
 
-    public QueryLifeTimeHookContextImpl build() {
+    public QueryLifeTimeHookContextImpl build(String queryId) {
       QueryLifeTimeHookContextImpl queryLifeTimeHookContext = new QueryLifeTimeHookContextImpl();
       queryLifeTimeHookContext.setHiveConf(this.conf);
       queryLifeTimeHookContext.setCommand(this.command);
       queryLifeTimeHookContext.setHookContext(this.hc);
+      queryLifeTimeHookContext.queryId = Objects.requireNonNull(queryId);
```

Comment:
We can enforce a min length check but its not easy to have a max since the query id includes the user name. The username max length is OS dependent.

I added a few more checks in https://github.com/apache/hive/pull/3156/commits/aae20a1fd3e9ae289daa3e6c95c20ae8a209211f. These are mostly sanity checks. I don't think we should put very involved checks here. Asserting things for the shape of the query id should be done via unit tests.

Reply:
Yes, sanity check was the only ask.  Changes LGTM.